src/wui/game_objectives_menu.cc

```diff
@@ -18,94 +18,411 @@
 
 #include "wui/game_objectives_menu.h"
 
+#include "base/time_string.h"
+#include "graphic/font_handler.h"
+#include "graphic/style_manager.h"
+#include "graphic/text_layout.h"
 #include "logic/game_data_error.h"
 #include "logic/objective.h"
 #include "logic/player.h"
+#include "logic/playersmanager.h"
 #include "wui/interactive_player.h"
 
-#define BUTTON_HEIGHT 20
-#define OBJECTIVE_LIST 120
-#define FULL_OBJECTIVE_TEXT 240
+constexpr int16_t kSpacing = 4;
+constexpr int16_t kRowSize = 32;
+constexpr int16_t kButtonWidth = 128;
 
-inline InteractivePlayer& GameObjectivesMenu::iplayer() const {
-	return dynamic_cast<InteractivePlayer&>(*get_parent());
+DiplomacyConfirmWindow::DiplomacyConfirmWindow(InteractivePlayer& parent,
```

Comment:
I think it would be more fitting if the diplomacy window were completely separate from the objectives menu. I don't think they are closely related to each other.

A "game session settings" menu might be good, it could contain the diplomacy menu and game speed setting, for example?